citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -3989,7 +4016,44 @@ public void doImportCitationFromResourceUrl( RunData data )
 	    setMode(state, Mode.LIST);
 
 	} // doImportCitationsFromResourceUrl
-	
+
+	private Citation extractCitationFromRunData(RunData data) {
+		try {
+			SessionState session = ((JetspeedRunData)data).getPortletSessionState(((JetspeedRunData) data).getJs_peid());
+			if(session.getAttribute(FilePickerHelper.FILE_PICKER_ATTACHMENTS) != null) {
+				List refs = (List) session
+						.getAttribute(FilePickerHelper.FILE_PICKER_ATTACHMENTS);
+				String citationCollectionId = (String) sessionManager.getCurrentToolSession().getAttribute(STATE_CITATION_COLLECTION_ID);
+				CitationCollection collection = getCitationService().getCollection(citationCollectionId);
+				Citation citation = null;
+				if (!refs.isEmpty()) {
+					for (Iterator it = refs.iterator(); it.hasNext(); ) {
```

Comment:
That list is always expected to have only one item , as the file picker selection cardinality. I've removed the iteration and the unused collection variables